File: 프로그래머스/LV1/코딩테스트 연습/연습문제/카드 뭉치/카드 뭉치.py
from collections import deque

def solution(cards1, cards2, goal):
    q1 = deque(cards1)
    q2 = deque(cards2)
    goal_idx = 0
    
    while q1 or q2:
        if goal_idx == len(goal): break
        
        if q1 and q1[0] == goal[goal_idx]:
            q1.popleft()
            goal_idx += 1
        elif q2 and q2[0] == goal[goal_idx]:
            q2.popleft()
            goal_idx += 1
        else:
            break
               
    return "Yes" if goal_idx == len(goal) else "No"

File: 프로그래머스/LV1/코딩테스트 연습/연습문제/카드 뭉치/카드 뭉치.cpp
#include <string>
#include <vector>
#include <queue>
using namespace std;

string solution(vector<string> cards1, vector<string> cards2, vector<string> goal) {
    queue<string> q1, q2;
    for (string c : cards1) q1.push(c);
    for (string c : cards2) q2.push(c);
    int goalIdx = 0;

    while (!q1.empty() || !q2.empty()) {
        if (goalIdx == goal.size()) break;

        if (!q1.empty() && q1.front() == goal[goalIdx]) {
            q1.pop();
            goalIdx++;
        }
        else if (!q2.empty() && q2.front() == goal[goalIdx]) {
            q2.pop();
            goalIdx++;
        }
        else break;
    }

    return goalIdx == goal.size() ? "Yes" : "No";
}